src/CTDeployer4_1.sol

```diff
@@ -0,0 +1,367 @@
+        // If the address is a sucker for this project.
+        return SUCKER_REGISTRY.isSuckerOf(projectId, addr);
+    }
+
+    /// @dev Make sure only mints can be received.
+    function onERC721Received(
+        address operator,
+        address from,
+        uint256 tokenId,
+        bytes calldata data
+    )
+        external
+        view
+        returns (bytes4)
+    {
+        data;
+        tokenId;
+        operator;
+
+        // Make sure the 721 received is the JBProjects contract.
+        if (msg.sender != address(PROJECTS)) revert();
+        // Make sure the 721 is being received as a mint.
+        if (from != address(0)) revert();
+        return IERC721Receiver.onERC721Received.selector;
+    }
+
+    //*********************************************************************//
+    // -------------------------- public views --------------------------- //
+    //*********************************************************************//
+
+    /// @notice Indicates if this contract adheres to the specified interface.
+    /// @dev See `IERC165.supportsInterface`.
+    /// @return A flag indicating if the provided interface ID is supported.
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return interfaceId == type(ICTDeployer4_1).interfaceId || interfaceId == type(IJBRulesetDataHook).interfaceId
+            || interfaceId == type(IERC721Receiver).interfaceId;
+    }
+
+    //*********************************************************************//
+    // ---------------------- external transactions ---------------------- //
+    //*********************************************************************//
+
+    /// @notice Deploy a simple project meant to receive posts from Croptop templates.
+    /// @param owner The address that'll own the project.
+    /// @param projectConfig The configuration for the project.
+    /// @param suckerDeploymentConfiguration The configuration for the suckers to deploy.
+    /// @param controller The controller that will own the project.
+    /// @return projectId The ID of the newly created project.
+    /// @return hook The hook that was created.
+    function deployProjectFor(
+        address owner,
+        CTProjectConfig calldata projectConfig,
+        CTSuckerDeploymentConfig calldata suckerDeploymentConfiguration,
+        IJBController controller
+    )
+        external
+        returns (uint256 projectId, IJB721TiersHook hook)
+    {
+        if (controller.PROJECTS() != PROJECTS) revert();
+
+        JBPayDataHookRulesetConfig[] memory rulesetConfigurations = new JBPayDataHookRulesetConfig[](1);
+        rulesetConfigurations[0].weight = 1_000_000 * (10 ** 18);
+        rulesetConfigurations[0].metadata.baseCurrency = uint32(uint160(JBConstants.NATIVE_TOKEN));
+
+        // Deploy a blank project.
+        // slither-disable-next-line reentrancy-benign
+        (projectId, hook) = DEPLOYER.launchProjectFor({
```

Comment:
ugghh. @xBA5ED we f'd up the croptop deployer. this should work like JBOmnichainDeployer4_1, using `deployHookFor` not `launchProjectFor`, then using the hook's address to launch a project inline with the `dataHook` being address(this). As it stands, the `dataHook` is the NFTs address, so all the omnichain features of this contract dont get called.